packages/ui/src/views/unified-pipeline-studio/components/entity-form/unified-pipeline-studio-entity-form.tsx

```diff
@@ -244,28 +244,30 @@ export const UnifiedPipelineStudioEntityForm = (props: UnifiedPipelineStudioEnti
                 <RenderForm className="space-y-5" factory={inputComponentFactory} inputs={formDefinition} />
               )}
             </EntityFormLayout.Form>
-          </Inner>
+          </Body>
           <Footer>
-            <div className="flex gap-x-3">
-              <Button disabled={loading || !!error?.message} onClick={() => rootForm.submitForm()}>
-                Submit
-              </Button>
-              <Button variant="secondary" onClick={requestClose}>
-                Cancel
-              </Button>
-            </div>
-            {editStepIntention && (
-              <Button
-                variant="secondary"
-                iconOnly
-                onClick={() => {
-                  requestYamlModifications.deleteInArray({ path: editStepIntention.path })
-                  requestClose()
-                }}
-              >
-                <Icon name="trash" />
-              </Button>
-            )}
+            <ButtonGroup className="justify-between">
+              <div className="flex gap-x-3">
+                <Button disabled={loading || !!error?.message} onClick={() => rootForm.submitForm()}>
+                  Submit
+                </Button>
+                <Button variant="secondary" onClick={requestClose}>
+                  Cancel
+                </Button>
+              </div>
+              {!!editStepIntention && (
+                <Button
```

Comment:
Looks like this component will be "button" to a screen reader. Can we add `aria-label`?